backend/cmm_helpers.ml

```diff
@@ -3051,54 +3051,69 @@ module Generic_fns_tbl = struct
 
     let max_send = 20
 
-    let max_apply = 404
-
     let max_tuplify = 100
 
     let considered_as_small_threshold = 20
 
+    type classify =
+      | Small
+      | Big of int
+      | Evict
+
+    let classify_to_bool = function Small | Big _ -> true | Evict -> false
+
+    let keep l = if l <= considered_as_small_threshold then Small else Big l
+
     let is_curry (kind, arity, result) =
       if not (check_result result)
-      then false
+      then Evict
       else
         match kind with
         | Lambda.Tupled ->
           let l = len_arity arity in
-          2 <= l && l <= considered_as_small_threshold
+          if 2 <= l && l <= considered_as_small_threshold then keep l else Evict
         | Lambda.Curried { nlocal } ->
           let l = len_arity arity in
```

Comment:
Although it does not originally come from this PR, it seems to me that this will cause a function with 2 value parameters, and say, 100 unboxed float parameters, to be considered as `Small` if at any point we start pre-generating generic functions for other layouts than `value`.

Reply:
That's right - I've made it a bit more clear that we only cache generated functions involving boxed values